protocol/shannon/protocol.go

```diff
@@ -288,15 +304,18 @@ func (p *Protocol) getAppsUniqueEndpoints(
 		logger.Debug().Msg("processing app.")
 
 		session, err := p.FullNode.GetSession(ctx, serviceID, app.Address)
+		// error fetching a session:
+		// Log an error
+		// skip current app.
 		if err != nil {
-			logger.Error().Err(err).Msg("Internal error: error getting a session for the app. Service request will fail.")
-			return nil, fmt.Errorf("could not get the session for service %s app %s", serviceID, app.Address)
+			logger.Error().Err(err).Msg("Internal error: error getting a session for the app: skipping the app.")
```

Comment:
Can you `Msgf` to include the app address snd service id?

It'll b easy to read.

Ditto below.